pkg/dockerregistry/server/configuration/configuration.go

```diff
@@ -151,3 +249,165 @@ func pushEnv(environ []envVar) error {
 	}
 	return nil
 }
+
+func setDefaultMiddleware(config *configuration.Configuration) {
+	// Default to openshift middleware for relevant types
+	// This allows custom configs based on old default configs to continue to work
+	if config.Middleware == nil {
+		config.Middleware = map[string][]configuration.Middleware{}
+	}
+	for _, middlewareType := range []string{"registry", "repository", "storage"} {
+		found := false
+		for _, middleware := range config.Middleware[middlewareType] {
+			if middleware.Name != middlewareName {
+				continue
+			}
+			if middleware.Disabled {
+				log.Errorf("wrong configuration detected, openshift %s middleware should not be disabled in the config file", middlewareType)
+				middleware.Disabled = false
+			}
+			found = true
+			break
+		}
+		if found {
+			continue
+		}
+		config.Middleware[middlewareType] = append(config.Middleware[middlewareType], configuration.Middleware{
+			Name: middlewareName,
+		})
+	}
+	// TODO(legion) This check breaks the tests. Uncomment when the tests will be able to work with auth middleware.
+	/*
+		authType := config.Auth.Type()
+		if authType != middlewareName {
+			log.Errorf("wrong configuration detected, registry should use openshift auth controller: %v", authType)
+			config.Auth = make(configuration.Auth)
+			config.Auth[middlewareName] = make(configuration.Parameters)
+		}
+	*/
+}
+
+func getServerAddr(options configuration.Parameters) (registryAddr string, err error) {
+	// TODO: Deprecate this environment variable.
+	registryAddr = os.Getenv(DockerRegistryURLEnvVar)
+	if len(registryAddr) == 0 {
+		registryAddr = os.Getenv(OpenShiftDefaultRegistryEnvVar)
```

Comment:
This should be preferred over `DockerRegistryURLEnvVar`. Since the second is deprecated.

Reply:
@miminar This code was moved from the `auth.go` as is. I would prefer to refactor it as a follow up.